### ISSUE TITLE
examples/advanced/advancedGraphFitter usage of std::vector

### ISSUE THREAD

**Issue by greenwald** [bat/bat#46]
examples/advanced/advancedGraphFitter uses a std::vector of BCGraphFitter pointers. ROOT complains about this in uncompiled mode, since we have not (and cannot, I believe) enable the + option in the LinkDef file used to generate the ROOT dictionaries. The example runs fine when compiled (in ROOT).

If anyone knows the proper workaround. Please chime in.

**Comment by fredRos** [comment]
Which ROOT version did you use?

I cannot reproduce your problem with 9923765b324 and root 6.04.00. The
interpreted mode runs fine. But compilation does not work
```
info in <TUnixSystem::ACLiC>: creating shared library
/home/beaujean/workspace/bat/examples/advanced/advancedGraphFitter/./graphFitterAdvancedExample_C.so
In file included from input_line_12:9:
/home/beaujean/workspace/bat/examples/advanced/advancedGraphFitter/./graphFitterAdvancedExample.C:36:10:
fatal error:
      'BAT/BCAux.h' file not found
#include <BAT/BCAux.h>
 ```
@oschulz: we had these problems before 0.9.4, and it reinforces the need
to run the examples with `make distcheck`. Can you remember how you
solved that problem?

**Comment by greenwald** [comment]
I'm using 5.34/25. I think you need to test on ROOT 5. How much of the community has actually switched to 6?

**Comment by oschulz** [comment]
According to a ROOT team member I spoke to earlier this year, ROOT-6 is now baseline for most LHC experiments. Also, ROOT-6 is now officially the recommended production version.

**Comment by fredRos** [comment]
To keep it simple, let's just an array of size 4. This is basic C and should work in any ROOT version

**Comment by fredRos** [comment]
I fixed this in #98 and reactivated the test

**Comment by fredRos** [comment]
gi
